### ISSUE TITLE
Fix tests with giac>=1.9.0.35

### ISSUE THREAD

**Issue by antonio-rojas** [sagemath/sage#34926]
<div id="comment:0"></div>

A warning `argument is not an equation` has been removed in 1.9.0.35, which breaks a few tests.

Component: **packages: standard**

Author: **Antonio Rojas**

Reviewer: **Matthias Koeppe**

_Issue created by migration from https://trac.sagemath.org/ticket/34926_

**Comment by antonio-rojas** [comment]
Branch: **[u/arojas/fix_tests_with_1_9_0_35](https://github.com/sagemath/sagetrac-mirror/tree/u/arojas/fix_tests_with_1_9_0_35)**

**Comment by antonio-rojas** [comment]
Author: **Antonio Rojas**

**Comment by antonio-rojas** [comment]
Commit: **[`f28961e`](https://github.com/sagemath/sagetrac-mirror/commit/f28961e88271c6685f9d7e4697f36affe3bbe65c)**

**Comment by antonio-rojas** [comment]
<div id="comment:2"></div>

New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/f28961e88271c6685f9d7e4697f36affe3bbe65c"><code>f28961e</code></a></td><td><code>Fix tests with giac 1.9.0.35</code></td></tr></table>

**Comment by antonio-rojas** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -1 +1 @@
-
+A warning `argument is not an equation` has been removed in 1.9.0.35, which breaks a few tests.
``````

**Comment by kiwifb** [comment]
<div id="comment:3" align="right">comment:3</div>

Looks fairly trivial. Is this the only issue with 1.9.0-35? I have been on leave and working remotely which took it of my radar.

**Comment by mkoeppe** [comment]
<div id="comment:4" align="right">comment:4</div>

LGTM. The failure in `sage/schemes/elliptic_curves/hom_frobenius.py` seems unrelated

**Comment by antonio-rojas** [comment]
<div id="comment:5" align="right">comment:5</div>

Replying to [François Bissey](#comment%3A3):
> Looks fairly trivial. Is this the only issue with 1.9.0-35? I have been on leave and working remotely which took it of my radar.

Actually 1.9.0-35 is totally unusable (without patching) since it ships a broken function declaration in a header. This seems to be fixed in 1.9.0-37, and there are no further issues as far as Sage is concerned.

As a side note, I've had to patch it since a few releases ago to make it build with `-Werror=format-security` but this doesn't affect Sage (which doesn't set this flag).

**Comment by mkoeppe** [comment]
Reviewer: **Matthias Koeppe**

**Comment by tornaria** [comment]
FWIW, this works and is necessary for me, please PR for 9.8.

**Comment by tornaria** [comment]
I did a PR rebased on top of develop so it gets tested.

**Comment by mkoeppe** [comment]
I've removed the branch from the issue description, as it is replaced by https://github.com/sagemath/sage/pull/35068.

**Comment by mkoeppe** [comment]
Merged in 10.0.beta0
